watertap/examples/flowsheets/case_studies/groundwater_treatment/groundwater_treatment.py

```diff
@@ -0,0 +1,346 @@
+    m.fs.feed.conc_mass_comp[0, "arsenic"].fix(conc_mass_arsenic)
+    m.fs.feed.conc_mass_comp[0, "uranium"].fix(conc_mass_uranium)
+    m.fs.feed.conc_mass_comp[0, "nitrate"].fix(conc_mass_nitrate)
+    m.fs.feed.conc_mass_comp[0, "phosphates"].fix(conc_mass_phosphates)
+    m.fs.feed.conc_mass_comp[0, "iron"].fix(conc_mass_iron)
+    m.fs.feed.conc_mass_comp[0, "filtration_media"].fix(conc_mass_filtration_media)
+
+    solve(m.fs.feed)
+
+    # pump
+    m.fs.pump.load_parameters_from_database(use_default_removal=True)
+    m.fs.pump.lift_height.fix(30)  # head of 30 m to pump groundwater
+
+    # microbial battery
+    m.fs.micbatt.load_parameters_from_database(use_default_removal=True)
+
+
+def initialize_system(m):
+    seq = SequentialDecomposition()
+    seq.options.tear_set = []
+    seq.options.iterLim = 1
+    seq.run(m, lambda u: u.initialize())
+
+
+def solve(blk, solver=None, tee=False, check_termination=True):
+    if solver is None:
+        solver = get_solver()
+    results = solver.solve(blk, tee=tee)
+    if check_termination:
+        assert_optimal_termination(results)
+    return results
+
+
+def display_results(m):
+    print("\n+++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+    print("++++++++++++++++++++ DISPLAY RESULTS ++++++++++++++++++++")
+    print("+++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+
+    unit_list = ["feed", "pump", "micbatt"]
+    for u in unit_list:
+        m.fs.component(u).report()
+
+    print("\n+++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
+
+
+def add_costing(m):
+    source_file = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)),
+        "groundwater_treatment_case_study.yaml",
+    )
+
+    m.fs.costing = ZeroOrderCosting(default={"case_study_definition": source_file})
+
+    costing_kwargs = {"default": {"flowsheet_costing_block": m.fs.costing}}
+    m.fs.pump.costing = UnitModelCostingBlock(**costing_kwargs)
+    m.fs.micbatt.costing = UnitModelCostingBlock(**costing_kwargs)
+
+    m.fs.costing.cost_process()
+
+    m.fs.costing.add_electricity_intensity(m.fs.feed.properties[0].flow_vol)
+
+    m.fs.costing.spent_media_recovery_mass = Expression(
+        expr=m.fs.costing.utilization_factor
+        * pyunits.convert(
+            m.fs.byproduct.flow_mass_comp[0, "filtration_media"],
+            to_units=pyunits.kg / m.fs.costing.base_period,
+        ),
+        doc="Mass of spent filtration media generated per year (accounting for utilization factor)",
+    )
+
+    m.fs.costing.cost_spent_media_disposal = Expression(
+        expr=pyunits.convert(
+            m.fs.costing.spent_media_recovery_mass
+            * m.fs.costing.filtration_media_disposal_cost,
+            to_units=m.fs.costing.base_currency / m.fs.costing.base_period,
+        ),
+        doc="Disposal cost of spent filtration media per year (accounting for utilization factor)",
+    )
+
+    m.fs.costing.fresh_media_mass = Expression(
```

Comment:
I think here, or in `cost_fresh_media`, we can assign some separate utilization rate that accounts for the annual frequency at which fresh media is supplied (and paid for).

Reply:
The way it appears to me, it seems that fresh media is constantly provided throughout the year. I'm describing the addition of something like (e.g.,) 0.05 as the fraction of the operational time within a year that new media is actually provided... unless we are assuming that the quantity of annual fresh media specified is meant to be equivalent to what I'm describing.

On second thought, I think what you have here is sufficient for the initial implementation, and we can refine/augment as needed in subsequent PRs.

Reply:
The model makes it appear like fresh media is added continuously to the inlet stream, but I do not think that is how it would work in reality. Rather, fresh media would probably need to be dosed at discrete, periodic intervals. The amount of fresh media in the inlet stream is meant to provide the "average" amount of fresh media that enters the reactor over time. You can think of this as the model "annualizing" the addition of fresh media provided to the reactor.

We don't have the capability to model the kind of transient behavior that I think would occur in reality for this system. So the "steady-state" values that this model outputs aren't representative of the system at any single individual point in time. Rather, they represent the "average" steady-state over a year of typical operation. I do agree that we might want to refine this model later.